core/blockchain.go

```diff
@@ -1326,6 +1326,51 @@ func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.
 	return bc.writeBlockWithState(block, receipts, state)
 }
 
+func (bc *BlockChain) ReadAccumulatedEpochUptime(state *state.StateDB, epoch uint64) (*istanbul.Uptime, error) {
```

Comment:
up to this point the `Blockchain` type didn't know *anything* about uptimes. shouldn't this live somewhere else? I don't see how `uptimes` is a topic related to the `Blockchain` object. 

What do yo think @kevjue?

Reply:
Going to remove this function, I didn't use it at the end. They should probably be managed in the same place as the valset snapshot, which is currently at istanbul backend. But the relation between istanbul backend and chain objects is confusing, in some methods it's passed as argument, and others use the `chain` field.